src/dmd/sapply.d

```diff
@@ -170,6 +170,12 @@ public:
     {
         doCond(s.statement) || applyTo(s);
     }
+
+    override void visit(ForwardingStatement s)
+    {
+        if (s.statement)
+            doCond(s.statement) || applyTo(s.statement);
```

Comment:
I don't know the exact purpose of the walkPostorder visitor, but this seems fishy to me as ForwardingStatement is not supposed to have behaviour different from the wrapped statement.

Reply:
This is correct, as you want to walk over the wrapped statement, not the statement itself.

Reply:
If I have a ForwardingStatement wrapping a LabelStatement, doCond is called with different arguments than if I have only a LabelStatement. Before your changes, the behaviour was identical for both cases. Is this change intentional?

Reply:
Do you have code example that changes behaviour? Or is that just hypothetical?

The order that it walks the AST should be the same as the original.

Should probably add `applyTo(s)` if the underlying statement is null, but that does not happen in the current.

Reply:
Running the visitor on a semantically-analyzed version of the following code will result in changed behaviour:
```d
static foreach(i;0..1) L: writeln(i);
```
`L: writeln(i)` will be visited by the `StoppableVisitor v` twice now, even though before it was visited only once.

I don't think `applyTo(s)` should be called if `s.statement` is `null`.

Reply:
So then, `doCond(s.statement) || applyTo(s);`

Reply:
Only if it makes sense to Visit the ForwardingStatement by default. (I don't see why it would.)
Is this Visitor used for anything?

Reply:
It's used by `comeFrom` and/or `blockExit` for determining reachable statements.

Reply:
Visiting the ForwardingStatement will break the `hasCode` method.

Reply:
Right, that sounds like what I saw last night when seeing unreachable warnings.

So just the one `applyTo(s.statement` then to elide double visiting.

Reply:
I think it should just be
```
if(s.statement)
    s.statement.accept(this);
```
(Otherwise, e.g. bodies of while loops inside `static foreach` will not be visited.)